network.go

```diff
@@ -195,6 +195,12 @@ func (n *network) CreateEndpoint(name string, options ...EndpointOption) (Endpoi
 	if name == "" {
 		return nil, ErrInvalidName(name)
 	}
+
+	e, _ := n.EndpointByName(name)
```

Comment:
yes exactly. Since we agreed to change the signature of EndpointByName(), we'd better use it now.
Suggestion, if you collapse the the query with the if statement it will not look as awkward.